app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionaryAddWordContents.java

```diff
@@ -0,0 +1,363 @@
+            if (null == cursor) return false;
+            return cursor.getCount() > 0;
+        } finally {
+            if (null != cursor) cursor.close();
+        }
+    }
+
+    public static class LocaleRenderer {
+        private final String mLocaleString;
+        private final String mDescription;
+
+        public LocaleRenderer(final Context context, @Nullable final String localeString) {
+            mLocaleString = localeString;
+
+            if (null == localeString || "".equals(localeString)) {
+                mDescription = context.getString(R.string.user_dict_settings_all_languages);
+            // Special treatment for the known languages with _zz and _ZZ types
+            } else if (localeString.endsWith("_zz") || localeString.endsWith("_ZZ")) {
+                final int resId = context.getResources().getIdentifier("subtype_"+localeString, "string", context.getPackageName());
+                mDescription = context.getString(resId);
+            } else {
+                mDescription = LocaleUtils.constructLocaleFromString(localeString).getDisplayName();
+            }
+        }
+        @Override
+        public String toString() {
+            return mDescription;
+        }
+        public String getLocaleString() {
+            return mLocaleString;
+        }
+
+    }
+
+    private static void addLocaleDisplayNameToList(final Context context,
+            final ArrayList<LocaleRenderer> list, final String locale) {
+        if (null != locale) {
+            list.add(new LocaleRenderer(context, locale));
+        }
+    }
+
+    // Helper method to get the list of locales and subtypes to display for this word
+    public ArrayList<LocaleRenderer> getLocalesList(final Activity activity) {
+
+        final SharedPreferences prefs = DeviceProtectedUtils.getSharedPreferences(activity.getApplicationContext());
+        final boolean localeSystemOnly = prefs.getBoolean(Settings.PREF_USE_SYSTEM_LOCALES, true);
+        final ArrayList<LocaleRenderer> localesList = new ArrayList<>();
+
+        // List of main language
+        final List<InputMethodSubtype> enabledMainSubtype = SubtypeSettingsKt.getEnabledSubtypes(prefs, true);
+        // List of system language
+        final List<Locale> enabledSystemLocale = SubtypeSettingsKt.getSystemLocales();
+        // List of user dictionary
+        final TreeSet<String> userDictionaryList = UserDictionaryListFragment.getUserDictionaryLocalesSet(activity);
+        // To combine lists
+        final TreeSet<String> languageList = new TreeSet<>(String::compareToIgnoreCase);
+
+        // Add languages from the user dictionary list
+        // (local system languages and dictionary languages in which a word is already saved)
+        if (userDictionaryList != null) {
+            languageList.addAll(userDictionaryList);
+        }
+
+        // mLocale is removed from the language list as it will be added to the top of the list
+        languageList.remove(mLocale);
```

Comment:
You can do the removal after the set is filled, then you don't need to check for `mLocale` every time you add something to the set.

Reply:
Change made with commit 0582df5

Reply:
I meant to move it after the last call to `languageList.add`. But if it works here without issue, that's fine too.

Reply:
I can confirm that it works.
BTW, I had forgotten to remove "For all languages" as this is inserted at the end of the list.
It's fixed with the commit b8779cc